q2_types/_feature_data.py

```diff
@@ -153,6 +162,26 @@ def generator_to_dna_sequences(view, data_dir):
     skbio.io.write(view, format='fasta', into=file)
 
 
+def paired_dna_sequences_to_generator(data_dir):
+    left = skbio.io.read(os.path.join(data_dir, 'left-dna-sequences.fasta'),
+                         format='fasta', constructor=skbio.DNA)
+    right = skbio.io.read(os.path.join(data_dir, 'right-dna-sequences.fasta'),
+                          format='fasta', constructor=skbio.DNA)
+    for lseq, rseq in zip(left, right):
+        assert rseq.metadata['id'] == lseq.metadata['id'], \
+               lseq.metadata['id'] + ' and ' + rseq.metadata['id'] + ' differ'
+        yield lseq, rseq
+
+
+def generator_to_paired_dna_sequences(view, data_dir):
+    lfile = os.path.join(data_dir, 'left-dna-sequences.fasta')
+    rfile = os.path.join(data_dir, 'right-dna-sequences.fasta')
+
+    lview, rview = zip(*view)
```

Comment:
This will consume the generator and create two tuples holding all sequences in-memory. Here's a more efficient way to accomplish this, requiring only a single iteration through the generator and two sequences in memory at any given time:

```python
lfilepath = os.path.join(data_dir, 'left-dna-sequences.fasta')
rfilepath = os.path.join(data_dir, 'right-dna-sequences.fasta')

with open(lfilepath, 'w') as lfile, open(rfilepath, 'w') as rfile:
    for lseq, rseq in view:
        skbio.io.write(lseq, format='fasta', into=lfile)
        skbio.io.write(rseq, format='fasta', into=rfile)
```